var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -657,36 +658,35 @@ def __init__(self, spec, **kwargs):
         if '+float32' in spec:
             self.precision_option = 'SP'
 
-        # spec.architecture.platform is like `uname -s`, but lower-case
-        platform = spec.architecture.platform
+        # OpenFOAM chooses which wmake rules to use based on the output of
+        # `uname -s` and `uname -m`.  We can get the same information from
+        # Python's platform library.
 
-        # spec.architecture.target is like `uname -m`
-        target   = spec.architecture.target
+        system = platform.system().lower()
+        machine = platform.machine()
 
```

Comment:
I've never had the pleasure of needing to cross-compile something, so I'll let someone else answer this. I don't know enough about the difference between Spack's platform detection and the builtin platform module. @wscullin?